query/src/org/labkey/query/QueryServiceImpl.java

```diff
@@ -2740,6 +2742,19 @@ public void registerSchemaLinkAction(@NotNull Class<? extends Controller> action
         return schemaLinks;
     }
 
+    @Override
+    public void registerQueryIconURLProvider(QueryIconURLProvider queryIconProvider)
+    {
+        _queryIconURLProviders.add(queryIconProvider);
+    }
+
+    @Override
+    @NotNull public List<QueryIconURLProvider> getQueryIconURLProvider()
```

Comment:
```suggestion
    @NotNull public List<QueryIconURLProvider> getQueryIconURLProviders()
```
Returning a list, so plural seems appropriate